Samples/resource-owner-password-credential/Angular2/Angualar2-Client-ROPC/app/User/user-component.ts

```diff
@@ -0,0 +1,38 @@
+﻿import {Component} from '@angular/core'
+import {Router} from '@angular/router';
+import {ResourceService} from '../resource/resource-service';
+declare var System;
+
+@Component({
+    selector: 'authorize',
+    template: `<h1>You are logged in</h1> <h3><img style="vertical-align:middle" src="../../images/Preloader.gif" width="30" [hidden]="!isLoading" />{{payload}}</h3><hr/>
+               `,
+})
+
+export class userComponent {
+    private payload: string="";
+   public isLoading : boolean; // to show the loading progress
+    constructor( private _parentRouter: Router, 
+                 private resourceService:ResourceService) {}
+
+    ngOnInit() {
+        this.callResourceServer();
+    }
+
+    public callResourceServer() {
+        this.isLoading=true;
+        if( localStorage.getItem("auth_key")){
```

Comment:
As a design perspective, I feel that the AuthService is in charge of exposing a method to verify the token is there and still valid.
The token storage (via local storage) should be handled by another service in charge of storing and retriving tokens.